Automattic/wp-calypso
# Reader: Reimplement keyboard shortcuts - take 2

#### Changes proposed in this Pull Request

This PR aims to reintroduce a simpler version of the keyboard shortcuts that were supported in Reader and were removed in #60501. See https://github.com/Automattic/wp-calypso/pull/60501#issuecomment-1028083347 and https://github.com/Automattic/wp-calypso/pull/60501#issuecomment-1028593376 for the motivation.

The PR is essentially reverting parts of #60501 that were related to the Reader, but also update the keyboard shortcut handling to happen directly in the components, rather than using the legacy removed keyboard shortcuts module.

**This is a take 2 of #60812, which was reverted because of a bug (#61063). As a fix, we're disabling the keyboard shortcuts when triggered inside any kind of input field.**

#### Testing instructions

* Keep an eye on the console for errors while testing this.
* Go to `/read`
* Test the following keyboard shortcuts and verify they work well:
  * `j` - move selection down
  * `k` - move selection up
  * `Enter/Return` - open the full post view for the selected post
  * `v` - open the selected post in a new tab
  * `l` (lowercase L) - toggle like/unlike the selected post
  * `.` - go to top - works only if you have new updates in the stream.
* Open a selected post in a full post view
* Test the following keyboard shortcuts and verify they work well:
  * `Esc` - close full post view
  * `l` (lowercase L) - toggle like/unlike the post
  * `j` - previous post
  * `k` - next post
* Start writing a comment in a post that has comments enabled.
* Verify that the shortcuts don't work while the comment field is focused.
